File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        vector<pair<int,int>>v(arr.size());
        for(int i=0;i<arr.size();++i){
            int cnt = 0,num=arr[i];
            while(num){
                if(num & 1)
                    ++cnt;
                num = num>>1;
            }
            v[i].first = cnt;
            v[i].second = arr[i];
        }
        sort(begin(v),end(v));
        for( int i=0;i<arr.size();++i){
            arr[i]=v[i].second;
        }
        return arr;
    }
};

File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        temp = []
        for i in range(len(arr)):
            cnt = 0
            num = arr[i]
            while(num):
                if ( num & 1):
                    cnt+=1
                num = num>>1
            temp.append([cnt,arr[i]])
        temp.sort()
        for i in range(len(arr)):
            arr[i]= temp[i][1]
        return arr